src/templates/utils.js

```diff
@@ -0,0 +1,36 @@
+import tcomb from 'tcomb-validation';
+
+function toGetter(prop) {
+  const capsProp = prop.charAt(0).toUpperCase() + prop.slice(1);
+  return `get${capsProp}`;
+}
+
+/**
+ * Given a tcomb struct and a Javascript object used to instantiate that
+ * struct, create getters for all props on the tcomb struct and then
+ * instantiate it. This is intended to be used when constructing the
+ * LazyTemplateInterface, where somewhat stronger typing can propogate errors
+ * earlier.
+ *
+ * @param {LazyTemplateInterface} TemplateInterface
+ * @param {object} instance
+ * @return {LazyTemplateInterface} A lazy template interface extended with
+ * getters.
+ */
+export function constructWithGetters(TemplateInterface, instance) {
```

Comment:
constructWithGetters is kind of a goofy name IMO but I can't think of anything better...

Reply:
agreed